### PR TITLE
Blog app: add authorization rules

### PR DESCRIPTION
**Pull request by amejid**
## Milestone 10: Add authorization rules

In this PR, the following were implemented:-

- Added a role column to the users table
- Used cancancan gem to enforce authorization
- Provided Access Control according to the user's privilege

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -13,4 +13,8 @@ class User < ApplicationRecord
   def recent_posts
     posts.order(created_at: :desc).limit(3)
   end
+
+  def admin?
+    role == 'admin'
+  end
```
Comment (Lucas-Masaba) [inline app/models/user.rb:19]: - It was awesome that you created your admin method in this user model 🏄

**Review by Lucas-Masaba — APPROVED** [review]
# Hello @amejid and @Abdullah2213565  👋 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

almost-there

## Highlights 🍾 
- Linter checks passing ✔️ 
- Successfully installed and implemented cancancan gem 🏄 
- Awesome implemented destroy actions 🏂 

## Optional suggestions 💭 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏 💻 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Lucas-Masaba in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 1 above)
Comment (amejid) [reply app/models/user.rb:19]: Appreciated.

**Comment by amejid** [comment]
Thanks for reviewing and approving our work.

**Pull request merged**